core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java

```diff
@@ -239,12 +254,46 @@ public int getDeferredCount() {
     }
   }
 
+  @Override
+  public FateId create(FateKey fateKey) {
+    FateId fateId = fateIdGenerator.fromTypeAndKey(getInstanceType(), fateKey);
+    Pair<TStatus,Optional<FateKey>> statusAndKey = getStatusAndKey(fateId);
+    TStatus status = statusAndKey.getFirst();
+    Optional<FateKey> tFateKey = statusAndKey.getSecond();
+
+    // Case 1: Status of UNKNOWN means doesn't exist, so we can create
+    if (status == TStatus.UNKNOWN) {
+      create(fateId, fateKey);
+      // Case 2: Status is NEW so this is unseeded, we can return and allow the calling code
+      // to reserve/seed as long as the existing key is the same and not different as that would
+      // mean a collision
+    } else if (status == TStatus.NEW) {
+      Preconditions.checkState(tFateKey.isPresent(), "Tx key column is missing");
```

Comment:
I don't think there is a test for a collision between a random fate id and hash derived fate id.  Im 99.9% sure this code works, but was wondering if creating a test for that is worthwhile for detecting regressions.

Reply:
@keith-turner - I just added a test for collision with key generated fate id and random id

The test works by creating the transaction using the key and then the test deletes the key outside of fate (either out of accumulo table or replaces the ZK node with an empty key). After removing the key the transaction appears to the store just like a normal random transaction so the test then verifies the failure when calling create again and also ensures the transaction can still be reserved and used since it was essentially "converted" to a random transaction.